designs/2019-linter-options/README.md

```diff
@@ -0,0 +1,126 @@
+- Start Date: 2019-05-12
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# The `linterOptions` property in Config Files
+
+## Summary
+
+This RFC adds `linterOptions` property to config files. People can use config files (including shareable configs!) instead of CLI options in order to configure some linter behavior.
+
+## Motivation
+
+We cannot configure some linter's behavior with config files, especially, shareable configs. It's convenient if we can configure those in shareable configs.
+
+## Detailed Design
+
+This adds `linterOptions` property to config files with three properties.
+
+```js
+// .eslintrc.js
+module.exports = {
+    linterOptions: {
+        allowInlineConfig: true,              // Corresponds to --no-inline-config / `options.allowInlineConfig`
+        reportUnusedDisableDirectives: "off", // Corresponds to --report-unused-disable-directives / `options.reportUnusedDisableDirectives`
```

Comment:
Currently, `--allow-inline-config` and `--report-unused-disable-directives` are global settings. With this change, would they be implemented on a file-specific basis? (For example, if an `overrides` block contains `allowInlineConfig`, would inline config only be allowed for the files matching the overrides?)

Reply:
Yes, it becomes file-specific.